### PR TITLE
[release-v1.9 ]Run go mod edit golang x net 

### PR DESCRIPTION
**Pull request by matzew**
Running:
* `go mod edit -replace golang.org/x/net=golang.org/x/net@v0.17.0`
* `make RELEASE=1.9 generate-release`

However, _before_ I could _successfully_ run `make RELEASE...`, I had to call:
* `./hack/update-deps.sh`

Due to this:
```
make RELEASE=v1.9 generate-release
./openshift/release/generate-release.sh
=== Update Codegen for knative.dev/eventing-kafka-broker
--- Generating checksums for configmap _example keys
go: inconsistent vendoring in /home/matzew/go/src/knative.dev/eventing-kafka-broker:
	golang.org/x/net: is replaced in go.mod, but not marked as replaced in vendor/modules.txt

	To ignore the vendor directory, use -mod=readonly or -mod=mod.
	To sync the vendor directory, run:
		go mod vendor
make: *** [Makefile:69: generate-release] Error 1
```

### CODE REVIEW

File: vendor/knative.dev/eventing/pkg/scheduler/statefulset/scheduler.go [hunk 1]
```diff
@@ -272,41 +272,13 @@ func (s *StatefulSetScheduler) scheduleVPod(vpod scheduler.VPod) ([]duckv1alpha1
...
-					state.Pending[vpod.GetKey()] += p.VReplicas
-				
-					p.VReplicas = 0
-				}
-			}
-
-			if p.VReplicas > 0 {
-				placements = append(placements, *p)
+			if state.IsSchedulablePod(st.OrdinalFromPodName(p.PodName)) {
+				placements = append(placements, *p.DeepCopy())
```
Comment (pierDipi) [inline vendor/knative.dev/eventing/pkg/scheduler/statefulset/scheduler.go:281]: This change is not expected here

Comment (matzew) [reply vendor/knative.dev/eventing/pkg/scheduler/statefulset/scheduler.go:281]: see the above desc. for steps I did. If there is something wrong. Tell me.

**Comment by matzew** [comment]
@pierDipi this PR now also contains the code from this PR https://github.com/openshift-knative/eventing-kafka-broker/pull/857

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**